File: solutions/fsharp/darts/1/Darts.fs
module Darts

let score x y =
    let d n = pown (0. - n) 2
    let distance = sqrt (d x + d y)

    match distance with
    | d when d <= 1 -> 10
    | d when d <= 5 -> 5
    | d when d <= 10 -> 1
    | _ -> 0

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

distance :: Float -> Float -> Float
distance x y = sqrt $ x * x + y * y

data Position
  = Inner
  | Middle
  | Outer
  | Outside

position :: Float -> Position
position d
  | d <= 1 = Inner
  | d <= 5 = Middle
  | d <= 10 = Outer
  | otherwise = Outside

score :: Float -> Float -> Int
score x y =
  case pos of
    Inner -> 10
    Middle -> 5
    Outer -> 1
    Outside -> 0
  where
    pos = position $ distance x y
